docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,230 @@ sidebar_position: 7
 
 # Development Lifecycle
 
-To look at/download BobaBoard's code, check out our guide to [Setting Up your Coding Environment](/docs/engineering/start-developing/before-you-start).
-You might also be interested in [Git Notes](/docs/engineering/legacy-pages/git-notes)
+If you're new to collaborating on a code project, you may be wondering what that process looks like. This page gives an overview of the development lifecycle, explains key terminology, and guides you through the step-by-step process of contributing to BobaBoard's codebase.
```

Comment:
Small tidbit on tech writing: assume that readers will be skimming the document rather than reading it in full.

Lists are very helpful in this context. So in this case, I'd do:
```
If you're new to collaborating on a code project, you may be wondering what that process looks like. This page will:
- Give an overview of the development lifecycle
- Explain key terminology
- Guide you through the step-by-step process of contributing to BobaBoard's codebase.
```